update.go

```diff
@@ -205,3 +267,85 @@ other options are ignored.
 		return nil
 	},
 }
+
+func getBlkioWeightDeviceFromString(deviceStr []string) ([]specs.WeightDevice, error) {
+	var devs []specs.WeightDevice
+	for _, s := range deviceStr {
+		elems := regBlkioWeightDevice.FindStringSubmatch(s)
+		if elems == nil || len(elems) < 5 {
+			return nil, fmt.Errorf("invalid value for %s", s)
+		}
+		var dev specs.WeightDevice
+		var err error
+		var weight uint64
+		if dev.Major, err = strconv.ParseInt(elems[1], 10, 64); err != nil {
+			return nil, fmt.Errorf("invalid value for %s, err: %v", elems[1], err)
+		}
+		if dev.Minor, err = strconv.ParseInt(elems[2], 10, 64); err != nil {
+			return nil, fmt.Errorf("invalid value for %s, err: %v", elems[2], err)
+		}
+		if weight, err = strconv.ParseUint(elems[3], 10, 16); err != nil {
+			return nil, fmt.Errorf("invalid value for %s, err: %v", elems[3], err)
+		}
+		dev.Weight = u16Ptr(uint16(weight))
+		if elems[4] != "" {
+			if weight, err = strconv.ParseUint(elems[4], 10, 16); err != nil {
+				return nil, fmt.Errorf("invalid value for %s, err: %v", elems[4], err)
+			}
+			dev.LeafWeight = u16Ptr(uint16(weight))
+		}
+		devs = append(devs, dev)
+	}
+
+	return devs, nil
+}
+
+func getBlkioThrottleDeviceFromString(deviceStr []string) ([]specs.ThrottleDevice, error) {
+	var devs []specs.ThrottleDevice
+	for _, s := range deviceStr {
+		elems := regBlkioThrottleDevice.FindStringSubmatch(s)
+		if elems == nil || len(elems) < 4 {
+			return nil, fmt.Errorf("invalid value for %s", s)
+		}
+		var dev specs.ThrottleDevice
+		var err error
+		var rate uint64
+		if dev.Major, err = strconv.ParseInt(elems[1], 10, 64); err != nil {
+			return nil, fmt.Errorf("invalid value for %s, err: %v", elems[1], err)
+		}
+		if dev.Minor, err = strconv.ParseInt(elems[2], 10, 64); err != nil {
+			return nil, fmt.Errorf("invalid value for %s, err: %v", elems[2], err)
+		}
+		if rate, err = strconv.ParseUint(elems[3], 10, 64); err != nil {
+			return nil, fmt.Errorf("invalid value for %s, err: %v", elems[3], err)
+		}
+		dev.Rate = u64Ptr(rate)
+		devs = append(devs, dev)
+	}
+	return devs, nil
+}
+
+func convertSpecWeightDevices(devices []specs.WeightDevice) []*configs.WeightDevice {
+	var cwds []*configs.WeightDevice
+	for _, d := range devices {
+		wd := &configs.WeightDevice{}
+		wd.Major = d.Major
+		wd.Minor = d.Minor
+		wd.Weight = *d.Weight
+		wd.LeafWeight = *d.LeafWeight
+		cwds = append(cwds, wd)
+	}
+	return cwds
+}
+
+func convertSpecThrottleDevices(devices []specs.ThrottleDevice) []*configs.ThrottleDevice {
+	var ctds []*configs.ThrottleDevice
+	for _, d := range devices {
+		td := &configs.ThrottleDevice{}
+		td.Major = d.Major
+		td.Minor = d.Minor
+		td.Rate = *d.Rate
+		ctds = append(ctds, td)
+	}
+	return ctds
+}
```

Comment:
You can use `configs.NewWeightDevice` and `configs.NewThrottleDevice` instead of these.